### DISCUSSION TITLE
Converting ANSI output from Jupyter Notebooks using MyST-NB

### DISCUSSION THREAD

**Discussion by thiippal** [executablebooks/meta#298] [Q&A]
Hey!

I'm using MyST-NB for my [openly available learning materials on applied language technology](https://applied-language-technology.readthedocs.io/).

Right now I'm working on [a section](https://applied-language-technology.readthedocs.io/en/latest/notebooks/part_iii/02_pattern_matching.html#examining-matches-in-context-using-concordances) where I use a small library named [wasabi](https://github.com/ines/wasabi) to highlight output using ANSI colours.

I have added [the following line to my conf.py](https://github.com/Applied-Language-Technology/website/blob/master/source/conf.py#L81):

```
nb_render_text_lexer = "myst-ansi"
```

Based on the documentation, I was under the impression that this would render ANSI outputs as they appear in the Jupyter Notebook. However, I can't get this to work. Am I calling the ANSI lexer correctly?

If you can help me out with this, I promise to submit a pull request to improve the MyST-NB documentation for this part. ✌️

**Comment by choldgraf** [comment]
Hmmm [the ansi outputs docs](https://myst-nb.readthedocs.io/en/latest/use/formatting_outputs.html) I would have thought that this would “just work”. What kernel are you using for the notebook? Maybe it is something non standard that’s outputting things in weird ways?

**Reply by thiippal** [reply]
To the best of my knowledge, wasabi printouts are styled using ANSI, but perhaps @ines can shine a light on this if she has time!

As for the kernel, here's my system information:
```
Python 3.8.5 (default, Jul 21 2020, 10:42:08) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.21.0 -- An enhanced Interactive Python. Type '?' for help.
```

A bit of extra information about Sphinx and MyST-NB config:

```
Running Sphinx v3.5.2
loading pickled environment... failed
failed: build environment version not current
myst v0.13.5: MdParserConfig(renderer='sphinx', commonmark_only=False, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, update_mathjax=True, enable_extensions=['dollarmath'], disable_syntax=[], url_schemes=None, heading_anchors=None, html_meta=[], footnote_transition=True, substitutions=[], sub_delimiters=['{', '}'])
```

**Comment by thiippal** [comment]
Hey @choldgraf any ideas how I should start debugging this? Still can't get it to work.

**Reply by choldgraf** [reply]
I'm not really sure as I've never used wasabi, but perhaps if you look at the structure of the output in the notebook it will help provide a clue? I'm not super sure how the ANSII parsing works so it's hard to help :-/ maybe @chrisjsewell has thoughts?
